src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -261,7 +267,8 @@ public static IHostBuilder AddScriptHostCore(this IHostBuilder builder, ScriptAp
                         {
                             // Go directly to the environment; We have no valid configuration from the customer at this point.
                             string appInsightsConnStr = Environment.GetEnvironmentVariable(EnvironmentSettingNames.AppInsightsConnectionString);
-                            RecordAndThrowExternalStartupException("Error building configuration in an external startup class.", ex, loggerFactory, appInsightsConnStr);
+                            string appInsightsAuthStr = Environment.GetEnvironmentVariable(EnvironmentSettingNames.AppInsightsAuthenticationString);
```

Comment:
This logic of checking two different ways of authing to app insights is repeated a few times. Is there a way we can extract into a common helper? Maybe one that optionally takes in `IConfiguration`, will try to extract the two values, then fallsback to `Environment` if `IConfiguration` was not provided, or it did not contain the necessary auth info.